src/txmempool.cpp

```diff
@@ -615,7 +615,7 @@ static void CheckInputsAndUpdateCoins(const CTransaction& tx, CCoinsViewCache& m
 
 void CTxMemPool::check(const CCoinsViewCache *pcoins) const
 {
-    LOCK(cs);
+    AssertLockHeld(cs);
```

Comment:
I don't think this change is necessary. `check()` isn't currently called anywhere with the lock held.